### ISSUE TITLE
applyBindings defect 

### ISSUE THREAD

**Issue by Strandedpirate** [knockout/knockout#441]
I ran into an issue were knockout 2.0 was throwing errors saying a
particular observableArray didn't exist in one of my view models for a
template I had setup. I have a page that consists of multiple view
models each applied to a particular DOM element. After hours of
troubleshooting it turned out that one of my calls to
ko.applyBindings() was sending a DOM element that no longer existed.
The class name had changed for that DOM element and we had missed
changing the call to applyBindings(). Knockout never
complains about sending an invalid selector to its applyBindings()
method it simply hummed along and presented a binding error with a
template that was 100% setup correctly and gave no indication to the
actual issue.

In short applyBindings should throw up when an invalid DOM element is
passed to it. At the very least this gives us run-time validation of
the setup instead of spending hours tracking unrelated errors down.

Here is the fiddle reproducing the issue: http://jsfiddle.net/StrandedPirate/MNMYx/2/

Error:
Unable to parse bindings.
Message: Reference Error: plans is not defined;
Bindings value: template: { name: 'modelATemplate', foreach: plans }
